code/game/objects/items/devices/portable_chem_mixer.dm

```diff
@@ -190,46 +199,51 @@
 				total_ph = rs.ph
 			if(is_hallucinating && prob(5))
 				chemname = "[pick_list_replacements("hallucination.json", "chemicals")]"
-			chemicals.Add(list(list("title" = chemname, "id" = ckey(temp.name), "volume" = total_volume, "pH" = total_ph)))
-	data["chemicals"] = chemicals
-	var/beakerContents[0]
+			chemicals += list(list("title" = chemname, "id" = temp.name, "volume" = total_volume, "pH" = total_ph))
+	.["chemicals"] = chemicals
+
+	.["isBeakerLoaded"] = beaker ? 1 : 0
```

Comment:
are these boolean values? they should be made so,

also this would be a very good spot for a discriminating union, if all of these are false if there is no beaker you could just return at `isBeakerLoaded` and create the discriminating type in TS